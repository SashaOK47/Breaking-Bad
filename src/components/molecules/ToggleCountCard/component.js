// import PropTypes from "prop-types";
import style from "./styles.module.scss";

const ToggleCountCard = ({ numbers, handleClick, itemsPerPage }) => {
  return (
    <div className={style.countCard}>
      <ul>
        {numbers.map((number) => {
          return (
            <li
              className={itemsPerPage == number ? style.active : null}
              key={number}
              id={number}
              onClick={handleClick}
            >
              {number}
            </li>
          );
        })}
      </ul>
    </div>
  );
};
export default ToggleCountCard;
