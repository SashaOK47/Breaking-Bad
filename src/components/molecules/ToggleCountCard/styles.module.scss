@import "../../../assets/styles/main.scss";
.countCard {
  .list {
    display: flex;
    align-items: center;
    @include tablet {
      flex-direction: column;
    }
    .text {
      margin-right: 1rem;
      @include tablet {
        margin-right: 0;
        margin-bottom: 1rem;
      }
    }
    button {
      @include filterBtn;
      padding: 0.25rem 0.5rem;
    }
  }
}
