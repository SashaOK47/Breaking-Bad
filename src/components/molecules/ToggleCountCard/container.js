import ToggleCountCard from "./component";
import { useSelector, useDispatch } from "react-redux";
import { UpdateItemsPerPage } from "../../../store/actions/characters";

const ToggleCountCardContainer = () => {
  const dispatch = useDispatch();
  const itemsPerPage = useSelector((state) => state.characters.itemsPerPage);
  const currentPage = useSelector((state) => state.characters.currentPage);
  const characters = useSelector((state) => state.characters.characters);

  let numbers = [];

  for (let i = 10; i <= 30; i++) {
    if (i % 5 == 0) {
      if (Math.ceil((currentPage - 1) * i) <= characters.length) {
        numbers.push(i);
      }
    }
  }

  const handleClick = (e) => {
    dispatch(UpdateItemsPerPage(Number(e.target.id)));
  };

  return (
    <ToggleCountCard
      numbers={numbers}
      handleClick={handleClick}
      itemsPerPage={itemsPerPage}
    />
  );
};
export const container = ToggleCountCardContainer;
