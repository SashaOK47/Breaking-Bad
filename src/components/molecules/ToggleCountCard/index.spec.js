import React from "react";
import ToggleCountCard from "./component";

const numbers = [10, 15, 20, 25];
const itemsPerPage = 15;
const handleClick = jest.fn();

const setUp = (props) => shallow(<ToggleCountCard {...props} />);

describe("ToggleCountCard component", () => {
  it("should render ToggleCountCard component", () => {
    const component = setUp({
      numbers,
      itemsPerPage,
      handleClick,
    });
    expect(component).toMatchSnapshot();
  });
});
