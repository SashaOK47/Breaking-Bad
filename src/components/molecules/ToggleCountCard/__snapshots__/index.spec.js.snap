// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleCountCard component should render ToggleCountCard component 1`] = `
<div
  className="countCard"
>
  <div
    className="list"
  >
    <span
      className="text"
    >
      Отобразить карточек: 
    </span>
    <div
      className="btns"
    >
      <button
        className={null}
        id={10}
        key="10"
        onClick={[MockFunction]}
      >
        10
      </button>
      <button
        className="active"
        id={15}
        key="15"
        onClick={[MockFunction]}
      >
        15
      </button>
      <button
        className={null}
        id={20}
        key="20"
        onClick={[MockFunction]}
      >
        20
      </button>
      <button
        className={null}
        id={25}
        key="25"
        onClick={[MockFunction]}
      >
        25
      </button>
    </div>
  </div>
</div>
`;
